Employee-Client/src/components/login/index.jsx

```diff
@@ -13,29 +12,49 @@ import Typography from '@mui/material/Typography';
 import Container from '@mui/material/Container';
 import { createTheme, ThemeProvider } from '@mui/material/styles';
 import { useEffect, useState } from 'react';
-// import customFetch from '../../utils/customFetch';
 import { useDispatch, useSelector } from "react-redux";
 import { loginUser } from '../../features/user/userSlice';
 import { useNavigate } from "react-router-dom";
+import useInput from '../../utils/use-input';
+import { getMsgErrorValidPass, isNotEmpty, validPass } from '../../utils/checkInputReg';
+import CustomInput from '../Input';
 
 
 const defaultTheme = createTheme();
 
 export function Login() {
-    const [username, setUsername] = useState('')
-    const [password, setPassword] = useState('')
+    const usernameInput = useInput(isNotEmpty);
+    const passwordInput = useInput(validPass);
     const dispatch = useDispatch();
+    const [submitIsValid, setSubmitIsValid] = useState(false);
     const { user } = useSelector((store) => store.user);
     const navigate = useNavigate();
+
+    const usernameInputErrorMsg = usernameInput.hasError
+        ? 'Username is required'
+        : '';
+
+    const userPasswordErrorMsg = passwordInput.hasError
+        ? getMsgErrorValidPass(passwordInput.value)
+        : '';
+
     const handleSubmit = (event) => {
         event.preventDefault();
-        // customFetch.post('/user/auth', { username: username, password: password })
-        // .then((res) => {
-        //     console.log(res);
-        // })
-        dispatch(loginUser({ username: username, password: password }));
+        dispatch(loginUser({ username: usernameInput.value, password: passwordInput.value }));
     };
 
+    const checkPgae = () => {
+        if (usernameInput.isValid && passwordInput.isValid) {
+            setSubmitIsValid(true)
+        } else {
+            setSubmitIsValid(false)
+        }
+    }
+
+    useEffect(() => {
+        checkPgae()
+    }, [usernameInput.value, passwordInput.value])
```

Comment:
Is it better to have 
```suggestion
    const checkPgae = () => {
        if (usernameInput.isValid && passwordInput.isValid) {
            setSubmitIsValid(true)
        } else {
            setSubmitIsValid(false)
        }
    }

    useEffect(() => {
        checkPgae()
    }, [usernameInput.isValid, passwordInput.isValid])
```

where `[usernameInput.value, passwordInput.value]` are not directly used in `checkPgae`